plugins/PrivacyManager/API.php

```diff
@@ -234,29 +228,113 @@ private function formatAvailableColumnsToAnonymize($columns)
+    {
+        if (Piwik::hasUserSuperUserAccess()) {
+            $jsCodeGenerator = new TrackerCodeGenerator();
+            $file = new File(PIWIK_DOCUMENT_ROOT . '/' . $jsCodeGenerator->getJsTrackerEndpoint());
+            $filename = $jsCodeGenerator->getJsTrackerEndpoint();
+
+            if (Manager::getInstance()->isPluginActivated('CustomJsTracker')) {
+                $file = StaticContainer::get('Piwik\Plugins\CustomJsTracker\TrackerUpdater')->getToFile();
+                $filename = $file->getName();
+            }
+
+            return [$filename, $file->hasWriteAccess()];
+        }
+
+        return ['', false];
+    }
+
+    /**
+     * Provide anonymisation settings to Matomo UI only
+     *
      * @internal
      */
-    public function setAnonymizeIpSettings($anonymizeIPEnable, $maskLength, $useAnonymizedIpForVisitEnrichment, $anonymizeUserId = false, $anonymizeOrderId = false, $anonymizeReferrer = '', $forceCookielessTracking = false, $randomizeConfigId = false)
+    public function getAnonymisationSettings(?int $idSiteSpecific = null): array
     {
-        Piwik::checkUserHasSuperUserAccess();
+        if (is_numeric($idSiteSpecific)) {
+            $idSite = intval($idSiteSpecific);
+            Piwik::checkUserHasAdminAccess($idSiteSpecific);
+        } else {
+            $idSite = null;
+            Piwik::checkUserHasSuperUserAccess();
+        }
+
+        $privacyConfig = new Config($idSite);
+        $settings = [];
+        foreach ($privacyConfig->getConfigPropertyNames() as $propertyName) {
+            $settings[$propertyName] = $privacyConfig->{$propertyName};
+        }
+        $settings['useSiteSpecificSettings'] = $privacyConfig->useSiteSpecificSettings();
+
+        // provide extra settings
+        [$trackerFilename, $trackerFileWritable] = $this->getTrackerFileDetails();
+        $settings = array_merge($settings, [
+            'maskLengthOptions' => PrivacyManager::getMaskLengthOptions(),
+            'useAnonymizedIpForVisitEnrichmentOptions' =>
+                PrivacyManager::getUseAnonymizedIpForVisitEnrichmentOptions(),
+            'referrerAnonymizationOptions' => ReferrerAnonymizer::getAvailableAnonymizationOptions(),
+            'configRandomisationFeatureFlag' =>
+                $this->featureFlagManager->isFeatureActive(ConfigIdRandomisation::class),
+            'trackerFileName' => $trackerFilename,
+            'trackerWritable' => $trackerFileWritable,
+        ]);
+
+        return $settings;
+    }
+
+    /**
+     * @internal
+     */
+    public function setAnonymisationSettings(
+        $enableIpAnonymizer,
+        $ipAddressMaskLength,
+        $useAnonymizedIpForVisitEnrichment,
+        $anonymizeUserId = false,
+        $anonymizeOrderId = false,
+        $anonymizeReferrer = '',
+        $forceCookielessTracking = false,
+        $randomizeConfigId = false,
+        $idSiteSpecific = null,
+        $useSiteSpecificSettings = false
```

Comment:
It would be beneficial to directly add type hints for all or at least for the new parameters when working on any API method. In this case it should be quite easy, as a parameter type hinted with boolean will also accept e.g. `0` or `1` .
And for `$idSiteSpecific` you could use `?int`, which would make the `is_numeric` check obsolete.

Reply:
Ok, that's fair. Since it's an internal API, I've added types to all the params and removed casting within the method where not needed anymore.